### PR TITLE
Enable Privileged Services

### PR DESCRIPTION
**Pull request by Nino-K**
This will allow us to start RD guest agent with the correct parameters when privileged service is installed.

Signed-off-by: Nino Kodabande <nkodabande@suse.com>

### CODE REVIEW

File: src/backend/wsl.ts [hunk 1]
```diff
@@ -691,21 +691,44 @@ export default class WSLBackend extends events.EventEmitter implements VMBackend
     await this.wslInstall(trivyExecPath, '/usr/local/bin');
   }
 
-  protected async installGuestAgent(kubeVersion: semver.SemVer | undefined) {
+  protected async installGuestAgent(kubeVersion: semver.SemVer | undefined, cfg: RecursivePartial<BackendSettings>) {
+    let guestAgentConfig: Record<string, any>;
+    let privilegedServiceEnabled;
```
Comment (ericpromislow) [inline src/backend/wsl.ts:696]: `privilegedServiceEnabled` is never set to true. Is there supposed to be a config on this?

File: src/backend/wsl.ts [hunk 2]
```diff
@@ -691,21 +691,44 @@ export default class WSLBackend extends events.EventEmitter implements VMBackend
...
+    const kube = K3sHelper.requiresPortForwardingFix(kubeVersion);
```
Comment (ericpromislow) [inline src/backend/wsl.ts:697]: Not part of this PR strictly speaking, but `kube` isn't a great name. Better to make this `addKubernetesSupport` or whatever `GUESTAGENT_KUBERNETES` means.

**Review by ericpromislow — CHANGES_REQUESTED** [review]
One rename request, one error.

**Comment by Nino-K** [comment]
@ericpromislow PTAL, I think our CI is acting up 🤔

**Review by ericpromislow — CHANGES_REQUESTED** [review]
lint error:
```
[error] ERROR in src/backend/wsl.ts:1140:120
TS2345: Argument of type 'RecursiveReadonly<RecursiveReadonly...
```

**Review by ericpromislow — APPROVED** [review]
It works now. Trying to run `privileged-service.exe start` fails; I assume that's out of scope for this PR.

**Comment by Nino-K** [comment]
https://github.com/rancher-sandbox/rancher-desktop/issues/2677

**Pull request merged**